src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -289,6 +289,11 @@ public async Task<object> GetDescriptionAsync(IAsyncCompletionSession session, V
             CompletionService completionService,
             Dictionary<string, AsyncCompletionData.CompletionFilter> filterCache)
```

Comment:
should we create new dictionary every time new completion is created? looks like it can be pooled?

Reply:
As mentioned in the PR description, I believe we can make completion perf a lot better by optimization the completion sub-system. However, I will not make that move for 16.1 preview 2 due to its potential impact.

That being said, this is a very good point and simple to fix. So.. done :)